### PR TITLE
[DottyGraph] Wrap very long node names when dumping graph.

### PR DESCRIPTION
**Pull request by mciprian13**
**Summary**
Wrap very long node names (longer than 100 chars) when dumping the graph. Such nodes make the graph harder to read.
One such example is this:
image
After wrapping the name the node looks like this:
image

### CODE REVIEW

**Comment by opti-mix** [comment]
@mciprian13 The idea makes thanks! Thanks! But could you make this configurable via a command-line option maybe? Also, have you considered multi-line names as described here: https://stackoverflow.com/questions/10841135/newline-in-node-label-in-dot-graphviz-language? This could be another way without loosing the information.

**Comment by mciprian13** [comment]
@opti-mix I'm not sure using multiple lines makes the thing much more readable. For my specific context I have very large node names. Wrapping the name would look like this:
```
Name : Relu6_model_init_conv_block_StatefulPartitionedCall_StatefulPartitionedCall_Relu6_batch_normalizati
       on_FusedBatchNormV3_model_init_conv_block_StatefulPartitionedCall_StatefulPartitionedCall_batch_nor
       malization_FusedBatchNormV3_batch_normalization_1_FusedBatchNormV3_model_encoder_sep_conv_block_0_0
       _StatefulPartitionedCall_StatefulPartitionedCall_batch_normalization_1_FusedBatchNormV3_depthwise_c
       onv2d_depthwise_model_encoder_sep_conv_block_0_0_StatefulPartitionedCall_StatefulPartitionedCall_tr
       unc_2__2
```
WDYT?

**Comment by opti-mix** [comment]
@mciprian13 I totally understand the motivation. We also see some very long names sometimes. 

The problem is that in some cases they differ only in the suffix part at the end. If we truncate unconditionally, some of these truncated names may result in the same name. This is why I was suggesting either having an option to disable truncation of names or using multi-lines.

**Comment by mciprian13** [comment]
@opti-mix I implemented a wrapping functionality which wraps the node name on multiple lines if longer than 100 characters. Find the example in the PR description. WDYT?

**Comment by jfix71** [comment]
Cool -- any idea if the name string is still searchable, at least when generated into an svg? I will often search through svgs/dot files for the exact name of some node, but guessing the line break will cause this to stop working.

File: include/glow/Support/Support.h [hunk 1]
```diff
@@ -65,6 +65,21 @@ Stream &operator<<(Stream &os, const llvm::ArrayRef<E> list) {
...
 }
 
+/// \returns a truncated version of the input string \p str to a maximum
+/// length of \p length. If the string effectively needs truncation then a
+/// given \p suffix will pe appended to the truncated string to signal that
+/// string was actually truncated.
+inline std::string truncateString(const std::string &str, size_t length,
+                                  const std::string &suffix = "...") {
+  return (str.size() > length) ? (str.substr(0, length) + suffix) : str;
+}
```
Comment (jfix71) [inline include/glow/Support/Support.h:75]: This one can be removed now since looks like it's unused?

Comment (mciprian13) [reply include/glow/Support/Support.h]: Done.

**Comment by mciprian13** [comment]
> Cool -- any idea if the name string is still searchable, at least when generated into an svg? I will often search through svgs/dot files for the exact name of some node, but guessing the line break will cause this to stop working.

@jfix71 Unfortunately by wrapping the node name you lose the capability to search it in svg/pdf. But you can still search it in the .dot file since there the DOT node instance name is the same as the unwrapped (original) node name.

**Pull request closed without merging**